integration-tools/src/test/java/uk/ac/bbsrc/tgac/miso/integration/test/dp5mirage/DP5MirageScannerTests.java

```diff
@@ -0,0 +1,137 @@
+package uk.ac.bbsrc.tgac.miso.integration.test.dp5mirage;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import uk.ac.bbsrc.tgac.miso.integration.BoxScan;
+import uk.ac.bbsrc.tgac.miso.integration.dp5mirage.DP5MirageScan;
+import uk.ac.bbsrc.tgac.miso.integration.dp5mirage.DP5MirageScanner;
+import uk.ac.bbsrc.tgac.miso.integration.dp5mirage.DP5MirageScanner.DP5MirageScanPosition;
+import uk.ac.bbsrc.tgac.miso.integration.test.BoxScannerTests;
+import uk.ac.bbsrc.tgac.miso.integration.util.IntegrationException;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+
+public class DP5MirageScannerTests extends BoxScannerTests<DP5MirageScanner> {
+  private static WireMockServer server;
+  private static DP5MirageScanner client;
+
+  @BeforeClass
+  public static void setup() throws IntegrationException {
+    server = new WireMockServer(WireMockConfiguration.wireMockConfig().dynamicPort());
+    server.start();
+    WireMock.configureFor("localhost", server.port());
+    client = new DP5MirageScanner("localhost", server.port());
+  }
+
+  @AfterClass
+  public static void shutdown() {
+    server.shutdown();
+  }
+
+  @Override
+  protected DP5MirageScanner getScanner() throws IntegrationException {
+    return client;
+  }
+
+  @Override
+  protected void simulateScan(BoxScan scan) {
+    // Return a specific canned HTTP response, depending on the barcode in first position [1,1]
```

Comment:
This is not what the method is intended to do. If the tests that use this are changed, these assumptions will fail. You are provided a `BoxScan`, which may contain any number of positions. Simulate that. It will be much easier to construct (and read/maintain) the JSON using Jackson classes. e.g.

```
ObjectMapper mapper = new ObjectMapper();
ObjectNode scan = mapper.createObjectNode();
scan.put("scanId", "3fa85f64-5717-4562-b3fc-2c963f66afa6");
...
int rows = scan.getRowCount();
int cols = scan.getColumnCount();
for (int col = 1; col <= cols; col++) {
  for (int row = 1; row <= rows; row++) {
    // create ObjectNode for position; add to "tubeBarcode" ArrayNode
  }
}
```